oneview_redfish_toolkit/blueprints/manager.py

```diff
@@ -91,3 +83,31 @@ def get_managers(uuid):
         # In case of error log exception and abort
         logging.exception('Unexpected error: {}'.format(e))
         abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+def _get_appliance_by_uuid(uuid, oneview_appliances):
+    appliance = dict()
+    appliance_index = 0
+
+    for index, ov_appliance in enumerate(oneview_appliances):
+        if ov_appliance["uuid"] == uuid:
+            appliance = ov_appliance
+            appliance_index = index
+            break
+
+    if not appliance:
+        raise OneViewRedfishError("Could not find manager with id " + uuid)
+
+    return appliance, appliance_index
+
+
+def get_current_manager():
+    oneview_appliances = \
+        g.oneview_client.appliance_node_information.get_version()
```

Comment:
We must avoid this overhead for each ResourceBlock. I believe we should cache appliance_uuid with the OV_ip during login and this will simplify a lot.

Reply:
Leveraging this thought, we can use `memoization` for this case, for example using [this](https://docs.python.org/3/library/functools.html#functools.lru_cache).